ladybug/windrose.py

```diff
@@ -0,0 +1,953 @@
+    @property
+    def legend(self):
+        """The Legend object for this plot"""
+        if self._legend is None:
+            self._legend = self._container._legend
+        return self._legend
+
+    @legend.setter
+    def legend(self, legend):
+        """Set the Legend object"""
+        assert isinstance(legend, type(Legend())), 'legend must be a Legend. ' \
+            'Got {}.'.format(type(legend))
+        self._legend = legend
+
+    @property
+    def direction_data_collection(self):
+        """The direction data collection assigned to this windrose plot."""
+        return self._direction_data_collection
+
+    @property
+    def analysis_data_collection(self):
+        """The analysis data collection assigned to this windrose plot."""
+        return self._analysis_data_collection
+
+    @property
+    def analysis_values(self):
+        """ The hourly wind analysis values to bin by direction (i.e wind speed)."""
+        return self._analysis_data_collection.values
+
+    @property
+    def direction_values(self):
+        """The direction data values in this windrose plot."""
+        return self._direction_data_collection.values
+
+    @property
+    def data(self):
+        """A histogram of wind analysis values binned by the wind directions."""
+        return self._data
+
+    @property
+    def zero_count(self):
+        """Number of wind analysis values equal to zero."""
+        return self._zero_count
+
+    @property
+    def analysis_period(self):
+        """The AnalysisPeriod assigned to the hourly plot's data collection."""
+        return self._analysis_data_collection.header.analysis_period
+
+    @property
+    def angles(self):
+        """List of orientation angles used to bin analysis values."""
+        return self._angles
+
+    @property
+    def bin_vectors(self):
+        """Vectors for orientation intervals."""
+        if self._bin_vectors is None:
+            self._bin_vectors = WindRose._bin_vectors_radial(self.angles)
+        return self._bin_vectors
+
+    @property
+    def zeros_per_bin(self):
+        """Number of analysis values that are equal to zero, per orientation bin."""
+        if self._zeros_per_bin is None:
+            self._zeros_per_bin = float(self._zero_count) / (len(self.angles) - 1)
+        return self._zeros_per_bin
+
+    @property
+    def real_freq_max(self):
+        """The maximum frequency of the windrose data."""
+        if self._real_freq_max is None:
+            self._real_freq_max = max([len(d) for d in self.data])
+        return self._real_freq_max
+
+    @property
+    def container(self):
+        """GraphicContainer for the windrose mesh."""
+        # Create the graphic container from the inputs and associated legend properties
+        max_pt = Point3D(self._center_point2d.x + self._radius,
+                         self._center_point2d.y + self._radius, self.base_point.z)
+
+        return GraphicContainer(self.analysis_values, self.base_point, max_pt,
```

Comment:
I think is something you should store as a hidden property on the class and you should set it to None/regenerate it whenever the `LegendParameters` or `base_point` are set.  Then, you can have the `WindRose.legend` as a property returned from this graphic container like you see on the `HourlyPlot` module.

Reply:
I'll check this with some unit tests tomorrow, but I think we complex datatypes like objects will be mutated within other objects, and we don't need to set it to None/regenerate whenever we change LegendParameters/base_point. If I'm wrong I'll modify to None.

Reply:
Nevermind. It looks like the radius is being used to intialize this object so my comment no longer applies (it's not just a object that sits within another object that chagnes, but the actual container class itself which will change.)